### PR TITLE
Add Commands onError hook

### PR DESCRIPTION
**Pull request by gonzalogarciajaubert**
## Description
Add Commands onError hook

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly

# Note: Base branch: feature/add_local_after_filter

### CODE REVIEW

**Review by rdiaz82 — APPROVED** [review]
LGTM

**Comment by javiertoledo** [comment]
I don't know if I see the value in this change. Can't you handle command errors using try/catch clauses already?

File: packages/framework-core/src/booster-command-dispatcher.ts [hunk 1]
```diff
@@ -38,14 +38,18 @@ export class BoosterCommandDispatcher {
...
+      await applyAfterFunctions(result, commandEnvelope.value, commandMetadata.after, register)
 
-    this.logger.debug('Command dispatched with register: ', register)
-    await RegisterHandler.handle(this.config, this.logger, register)
-    return result
+      this.logger.debug('Command dispatched with register: ', register)
+      await RegisterHandler.handle(this.config, this.logger, register)
+      return result
+    } catch (e) {
+      throw await applyOnErrorFunctions(e, commandEnvelope.value, commandMetadata.onError, register)
```
Comment (NickSeagull) [inline packages/framework-core/src/booster-command-dispatcher.ts:52]: Note that here `e` can be whatever, not only `Error`

**Review by NickSeagull — CHANGES_REQUESTED** [review]

**Comment by javiertoledo** [comment]
Closing, as this feature was replaced by the global error handler

**Pull request closed without merging**